test/basicfuns.jl

```diff
@@ -6,6 +6,11 @@ using StatsFuns, Test
 
     @test iszero(xlogy(0, 1))
     @test xlogy(2, 3) ≈ 2.0 * log(3.0)
+
+    @test xlogy(-2, Base.MathConstants.e) ≈ -2
+    @test xlogy(0, 1) ≈ 0
+    @test xlogy(-1, 1) ≈ 0
+    @test xlogy(0, 0) ≈ 0
```

Comment:
This isn't supposed to work according to the docstring. We could update it to say `y ≥ 0`, but then it wouldn't be right about the "correct limit" AFAICT. Maybe we should just drop that restriction and just say that zero is returned when `x == 0`. That's what [SciPy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.xlogy.html) does.

Reply:
I think the docstring is incorrect in this case. Maybe I'm misunderstanding the purpose of having a function like `xlogy`, but being able to evaluate 0log(0) is I believe intended to be its principal benefit.

I'm not sure I understand what you mean when you say,

> ...but then it wouldn't be right about the "correct limit" AFAICT

could you expand on this?

I dont believe dropping the constraint on `y` entirely is correct, as it should error out when `y` is of type negative real, irrespective of what `x` is.

My appraisal of the situation is that the docstring should be changed to `y≥0`.

Reply:
changed docstring to `y≥0`.

Reply:
I meant that I don't think it's correct to say that the limit of `x * log(y)` when `x` tends towards zero is zero when `y == 0`, is it? `x * log(0)` isn't defined when `x > 0`, so it's not appropriate to speak about its limit at `0`.

Reply:
Ok I see what you're getting at now, thanks for clarifying.

So strictly speaking you're right, the limit would not be defined. However I think the practical benefits of having `xlogy` evaluate 0log(0) as if you were actually taking the limit xlog(x) as x->0⁺ outweigh the slight lapse in mathematical rigor.

The docstring, as it currently stands, is:

`Return `x * log(y)` for `y > 0` with correct limit at `x = 0`.`

The mention of the limit at `x=0` being entirely superfluous when `y>0`, since it's unnecessary to consider any limits in this context, exact evaluation is always well-defined.

And to reiterate, if it isn't the problem of 0log(0) which we're trying to resolve, then it's not clear to me that the function `xlogy` provides any benefit over just simply evaluating `x * log(y)`.

Reply:
Yes, it's not a big deal. Though we could also use a more explicit mention about what happens when `x == 0`.

Reply:
ok, added a more descriptive docstring.